unlock-app/src/hooks/useProvider.ts

```diff
@@ -96,7 +96,7 @@ export const useProvider = (config: any) => {
         )) as unknown as ethers.BrowserProvider
       } else {
         await switchBrowserProviderNetwork(networkId)
-        walletServiceProvider = new ethers.BrowserProvider(pr.provider, 'any')
+        walletServiceProvider = new ethers.BrowserProvider(window.ethereum!)
```

Comment:
Are you sure this works with wallet connect or Coinbase wallet?

Reply:
for coinbase it should https://docs.cdp.coinbase.com/wallet-sdk/docs/injected-provider/